File: FinancialPlanning/FinancialPlanningAPI/Models/TermViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace FinancialPlanningAPI.Models
{
    public class TermViewModel
    {
        public Guid Id { get; set; }

        [Required(ErrorMessage = "Term name is required")]
        public string TermName { get; set; }

        [Required(ErrorMessage = "Creator ID is required")]
        public Guid CreatorId { get; set; }

        [Required(ErrorMessage = "Duration is required")]
        [RegularExpression(@"^[1,3,6]$", ErrorMessage = "Duration must be set to 1, 3, or 6")]
        public int Duration { get; set; }

        [Required(ErrorMessage = "Start date is required")]
        public DateTime StartDate { get; set; }

        [Required(ErrorMessage = "Plan due date is required")]
        public DateTime PlanDueDate { get; set; }

        [Required(ErrorMessage = "Report due date is required")]
        public DateTime ReportDueDate { get; set; }

        // [Required(ErrorMessage = "Status is required")]
        // [Range(1, 3, ErrorMessage = "Status must be set to 1, 2, or 3")]
        public int Status { get; }
        public TermViewModel(Guid id, string termName, Guid creatorId, int duration, DateTime startDate, DateTime planDueDate, DateTime reportDueDate, int status)
        {
            Id = id;
            TermName = termName;
            CreatorId = creatorId;
            Duration = duration;
            StartDate = startDate;
            PlanDueDate = planDueDate;
            ReportDueDate = reportDueDate;
            Status = status;
        }
    }
}

File: FinancialPlaningUI/FinancialPlaning/src/app/models/termview.model.ts
export class TermViewModel {
  id: string;
  termName: string;
  creatorId: string;
  duration: number;
  startDate: Date;
  planDueDate: Date;
  reportDueDate: Date;
  status: number;
  user: any; // You might want to define a UserViewModel here if needed

  constructor(data: any = {}) {
    this.id = data.id || '';
    this.termName = data.termName || '';
    this.creatorId = data.creatorId || '';
    this.duration = data.duration || 0;
    this.startDate = new Date(data.startDate);
    this.planDueDate = new Date(data.planDueDate);
    this.reportDueDate = new Date(data.reportDueDate);
    this.status = data.status || 0;
    this.user = data.user || null;
  }
}
